### PR TITLE
Support multiple authorization header for REST API

### PR DESCRIPTION
**Pull request by l2m83**
Manages multiple autorization header for REST API
Useful when there is an HTTP server authentication as well as the token for REST API.
https://mantisbt.org/bugs/view.php?id=25362

### CODE REVIEW

File: api/rest/restcore/AuthMiddleware.php [hunk 1]
```diff
@@ -51,14 +51,27 @@ public function __invoke( \Slim\Http\Request $request, \Slim\Http\Response $resp
 			}
 		} else {
 			# TODO: add an index on the token hash for the method below
-			$t_user_id = api_token_get_user( $t_authorization_header );
+
+			# Manage multiple authorization header (ex: Basic + token)
+			$authStringArray = explode(', ', $t_authorization_header);
+			$t_user_id = "";
+			$real_auth_header = "";
+			#Search for the token among the different authStrings
```
Comment (vboctor) [inline api/rest/restcore/AuthMiddleware.php]: Replace `#Search for the token among the different authStrings ` with `# Search for the token among the different authorization headers`.  Note the space and changing of authStrings string.

File: api/rest/restcore/AuthMiddleware.php [hunk 2]
```diff
@@ -51,14 +51,27 @@ public function __invoke( \Slim\Http\Request $request, \Slim\Http\Response $resp
...
+
+			# Manage multiple authorization header (ex: Basic + token)
+			$authStringArray = explode(', ', $t_authorization_header);
+			$t_user_id = "";
+			$real_auth_header = "";
+			#Search for the token among the different authStrings 
+			foreach ($authStringArray as $value) {
+	                        $t_user_id = api_token_get_user( $value );
+				if( $t_user_id !== false ) {
+					#Valid token found
```
Comment (vboctor) [inline api/rest/restcore/AuthMiddleware.php]: Add space after `#`.

File: api/rest/restcore/AuthMiddleware.php [hunk 3]
```diff
@@ -51,14 +51,27 @@ public function __invoke( \Slim\Http\Request $request, \Slim\Http\Response $resp
...
+					$real_auth_header = $value;
+					break;
+				}
+			}
```
Comment (vboctor) [inline api/rest/restcore/AuthMiddleware.php]: Add a blank line after `}`.

File: api/rest/restcore/AuthMiddleware.php [hunk 4]
```diff
@@ -51,14 +51,27 @@ public function __invoke( \Slim\Http\Request $request, \Slim\Http\Response $resp
 			}
 		} else {
 			# TODO: add an index on the token hash for the method below
-			$t_user_id = api_token_get_user( $t_authorization_header );
+
+			# Manage multiple authorization header (ex: Basic + token)
+			$authStringArray = explode(', ', $t_authorization_header);
```
[the remark this answers is not in the document]
Comment (l2m83) [reply api/rest/restcore/AuthMiddleware.php]: Change done

File: api/rest/restcore/AuthMiddleware.php [hunk 5]
```diff
@@ -51,14 +51,27 @@ public function __invoke( \Slim\Http\Request $request, \Slim\Http\Response $resp
...
+			$t_user_id = "";
+			$real_auth_header = "";
```
[the remark this answers is not in the document]
Comment (l2m83) [reply api/rest/restcore/AuthMiddleware.php]: Change done

File: api/rest/restcore/AuthMiddleware.php [hunk 6]
```diff
@@ -51,14 +51,27 @@ public function __invoke( \Slim\Http\Request $request, \Slim\Http\Response $resp
 			}
 		} else {
 			# TODO: add an index on the token hash for the method below
-			$t_user_id = api_token_get_user( $t_authorization_header );
+
+			# Manage multiple authorization header (ex: Basic + token)
+			$authStringArray = explode(', ', $t_authorization_header);
+			$t_user_id = "";
```
[the remark this answers is not in the document]
Comment (l2m83) [reply api/rest/restcore/AuthMiddleware.php]: Change done

File: api/rest/restcore/AuthMiddleware.php (hunk 1 above)
Comment (l2m83) [reply api/rest/restcore/AuthMiddleware.php]: Change done

File: api/rest/restcore/AuthMiddleware.php (hunk 2 above)
Comment (l2m83) [reply api/rest/restcore/AuthMiddleware.php]: Change done

File: api/rest/restcore/AuthMiddleware.php (hunk 3 above)
Comment (l2m83) [reply api/rest/restcore/AuthMiddleware.php]: Change done

**Comment by vboctor** [comment]
Thanks @l2m83 - merged via 8e84c922cea6a142d1a28210539a51fcc8c7ec20

**Pull request closed without merging**